de.seronet_projekt.ros.componentGateway.generator/src/de/seronet_projekt/ros/componentGateway/generator/ComponentGatewayGenerator.xtend

```diff
@@ -48,16 +48,16 @@ RosInterfacesPool {
 	RosPublisher «checkname(sub.name)»_pub { topicName "«sub.name»" type "«sub.subscriber.message.package.name».«sub.subscriber.message.name»" }
 	«ENDFOR»
 	«FOR srvserver:componentinterface.rosserviceserver»
-	RosSrvServer «checkname(srvserver.name)»_srvser { srvName "«srvserver.name»" type "«srvserver.srvserver.service.package.name».«srvserver.srvserver.service.name»" }
+	RosSrvServer «checkname(srvserver.name)»_srvcli { srvName "«srvserver.name»" type "«srvserver.srvserver.service.package.name».«srvserver.srvserver.service.name»" }
```

Comment:
What is the reasoning behind naming a server as a client and vice versa?

Reply:
this was **your** decision for the rosinterfacespool: mirrow the original ros interfaces.
See your comment here: 

https://github.com/seronet-project/SeRoNet-Tooling-ROS-Mixed-Port-intern/issues/16#issuecomment-442404045

> I just now realized that they all have the same "communication side" (I mean for the Joystick example it is always the "Publisher"), but the ROS MixedPort Component requires the other communication side (in this case the Subscriber).

and https://github.com/seronet-project/SeRoNet-Tooling-ROS-Mixed-Port-intern/pull/19

if for a Publisher is a Subscriber for a ServiceServer is a ServiceClient and viceversa.

Reply:
Yes, but in this case also the front string, e.g. the RosSrvServer needs to be changed to RosSrvClient and so forth, just like it is done for the Pub/Sub (there it is correct), otherwise this is inconsistent.

Reply:
Ah ok, I understand 
Yes you are right, I forgot that change

Reply:
@alex-lotz hope now it is right